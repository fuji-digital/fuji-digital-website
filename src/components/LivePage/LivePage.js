import {
  Header,
  PageStyling,
  Body,
  BlobStyling,
  CardLayout,
  DiscussingImage,
  TextContainer,
  HashtagTag,
  StyledContactButton,
} from "./LivePage.styles";

import { Logo } from "../Logo/Logo";
import { Card } from "../Card/Card";
import { Footer } from "../Footer/Footer";

import discussingImage from "../../images/decorative-images/discussing.svg";
import pinkPurpleDecorativeBlobs from "../../images/decorative-blobs/pink-purple-decorative-blobs.svg";

export const LivePage = () => {
  return (
    <PageStyling>
      <Header>
        <Logo version={"white"} />
      </Header>
      <BlobStyling src={pinkPurpleDecorativeBlobs} alt="A blue and pink blob" />
      <Body>
        <Card>
          <CardLayout>
            <HashtagTag>#flexiblesolutions </HashtagTag>
            <DiscussingImage
              src={discussingImage}
              alt="Two cartoon people discussing work over a computer"
            />
            <TextContainer>
              <h1>Where Ideas Take Digital Form</h1>
              <p>
                Lorem ipsum praesent ac massa at ligula reet est iaculis.
                Vivamus est mist aliquet elit ac nisl. Lorem ipsum praesent ac
                massa at ligula reet est iaculis. Vivamus est mist.
              </p>
            </TextContainer>
            <div style={{ display: "flex", justifyContent: "center" }}>
              <StyledContactButton text="Contact Us" />
            </div>
          </CardLayout>
        </Card>
      </Body>
      <Footer />
    </PageStyling>
  );
};
