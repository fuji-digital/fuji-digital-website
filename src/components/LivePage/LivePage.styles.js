import styled from "styled-components";
import { ContactButton } from "../ContactButton/ContactButton.js";
import { Modal } from "../Modal/Modal.js";

export const PageStyling = styled.div`
  position: relative;
  overflow-x: hidden;

  h1 {
    font-size: 1.5rem;
    font-family: "Poppins", sans-serif;
    color: #ebf0f7;
    line-height: 1.2;
    @media screen and (min-width: 1200px) {
      font-size: 1.8rem;
    }
  }

  h3 {
    color: #ebf0f7;
    font-family: "Poppins", sans-serif;
    margin: 0;
    font-size: 1.5rem;
  }

  p {
    font-size: 1rem;
    font-family: "Poppins", sans-serif;
    color: #ebf0f7;
    line-height: 1.2;
  }
`;

export const Body = styled.div`
  padding: 0rem 2.5rem 2.5rem 2.5rem;

  @media screen and (min-width: 768px) {
    padding: 0rem 2rem 2rem 2rem;
  }
`;

export const CardLayout = styled.div`
  display: flex;
  flex-direction: column;
  @media screen and (min-width: 768px) {
    display: grid;
    grid-template-columns: 1fr 2fr;
    grid-template-rows: 1fr 3fr 1fr;
    column-gap: 10px;
  }
`;

export const HashtagTag = styled.div`
  background-color: white;
  width: fit-content;
  height: fit-content;
  font-size: 0.9rem;
  border-radius: 50px;
  text-align: center;
  padding: 0.2rem 1rem;
  @media screen and (min-width: 768px) {
    grid-row: 1;
    grid-column: 1;
    align-self: end;
  }
`;

export const DiscussingImage = styled.img`
  height: 11rem;
  @media screen and (min-width: 768px) {
    height: 30vw;
    grid-column: 2;
    grid-row: 1 / 4;
    align-self: center;
    justify-self: center;
  }
`;
export const TextContainer = styled.div`
  @media screen and (min-width: 768px) {
    grid-column: 1;
    grid-row: 2 / 3;
    padding: 0rem 0rem 1rem;
  }
`;

export const StyledContactButtonContainer = styled.div`
  display: flex;
  justify-content: center;
  @media screen and (min-width: 768px) {
    justify-content: left;
  }
`;

export const StyledContactButton = styled(ContactButton)`
  @media screen and (min-width: 768px) {
    grid-column: 1;
    grid-row: 3;
  }
`;

export const StyledModal = styled(Modal)`
  @media screen and (min-width: 768px) {
    width: 38rem;
  }
`;
