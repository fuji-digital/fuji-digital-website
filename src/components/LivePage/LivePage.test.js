import renderer from "react-test-renderer";
import { LivePage } from "./LivePage";

jest.mock(`@uidotdev/usehooks`, () => ({
  useWindowSize: () => ({ width: 550 }),
}));

describe("LivePage", () => {
  it("matches snapshot", () => {
    const component = renderer.create(<LivePage />).toJSON();
    expect(component).toMatchSnapshot();
  });
});
