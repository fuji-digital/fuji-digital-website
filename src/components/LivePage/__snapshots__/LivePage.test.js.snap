// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LivePage matches snapshot 1`] = `
<div
  className="sc-eqUzNf hoWQDF"
>
  <div
    className="sc-fqkwJk AxtGy"
  >
    <div
      className="sc-cwHqhk fuAgyt"
    >
      <img
        alt="Fuji Digital Logo"
        className="sc-dAlxHm iJGhYI"
        src="fuji-digital-logo.svg"
      />
      <img
        alt="Fuji Digital Text Logo"
        className="sc-jlZhRR eJgNbx"
        data-testid="textLogo"
        src="fuji-digital-text-logo-white.svg"
      />
    </div>
  </div>
  <img
    alt="A blue and pink blob"
    className="sc-dcJtft hJIFfI"
    src="pink-purple-decorative-blobs.svg"
  />
  <div
    className="sc-iGgVNO cPqGEC"
  >
    <div
      className="sc-jEABJj goYoVf"
    >
      <div
        className="sc-gsFSjX liFMLo"
      >
        <div
          className="sc-kAycRU fXqYlV"
        >
          #flexiblesolutions 
        </div>
        <img
          alt="Two cartoon people discussing work over a computer"
          className="sc-imWZod dxfuJQ"
          src="discussing.svg"
        />
        <div
          className="sc-jXbVAB bFzrpW"
        >
          <h1>
            Where Ideas Take Digital Form
          </h1>
          <p>
            Our software engineering contracting website connects businesses with top-tier engineers, offering customized, scalable solutions for projects, ensuring quality and efficiency.
          </p>
        </div>
        <div
          className="sc-dhKdPU fUtPml"
        >
          <button
            className="sc-aYaIB jGsPsI"
          >
            <div>
              Contact Us
            </div>
            <img
              alt="A black and white arrow"
              className="sc-gEvDqW dTdWeV"
              src="arrow-icon.svg"
            />
          </button>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-feUYzb iMKqHn"
  >
    <div
      className="sc-fUnNpA iSlNrG"
    >
      What We Do
    </div>
    <div
      className="sc-hzhKNl cLKVWC"
    >
      <div
        className="sc-fHjqbK doayJt"
      >
        <img
          className="sc-hmdnzv bKYQGT"
          src="pink-stripe-fujit.svg"
        />
        <p>
          At our software engineering company, we prioritize delivering top-tier work for every website we create. We tailor our solutions to meet the unique needs of each client, ensuring high-quality, functional, and visually appealing results.
        </p>
      </div>
      <div
        className="sc-jsJARu fxNeA"
      >
        <img
          className="sc-hmdnzv bKYQGT"
          src="purple-tall-fujit.svg"
        />
        <p>
          We utilize the latest technologies to stay ahead of industry trends. By leveraging cutting-edge tools and techniques, we guarantee efficient, innovative, and secure websites that perform optimally across all devices.
        </p>
      </div>
      <div
        className="sc-fHjqbK doayJt"
      >
        <img
          className="sc-hmdnzv bKYQGT"
          src="pink-round-short-fujit.svg"
        />
        <p>
          Our team prides itself on exceptional customer service and timely project completion. We maintain open communication, ensuring clients are informed and satisfied throughout the development process.
        </p>
      </div>
    </div>
    <div
      className="sc-bXCLgj ebGGNL"
    >
      <img
        src="aws-icon.svg"
      />
      <img
        src="react-js-icon.svg"
      />
      <img
        src="graphql-icon.svg"
      />
      <img
        src="node-js-icon.svg"
      />
    </div>
    <p>
      Our software stack includes React for dynamic front-end development, Node.js for robust back-end solutions, and JavaScript for seamless integration. We utilize AWS for reliable cloud services and GraphQL for efficient data management. Always eager to innovate, we continuously explore and adopt new technologies to enhance our offerings.
    </p>
  </div>
  <div
    className="sc-eeDSqt gewKpw"
  >
    <div
      className="sc-jEABJj goYoVf"
    >
      <div
        className="sc-bmzXxz cipIEs"
      >
        Who Are We?
      </div>
      <div
        className="sc-koYCI iKUMOe"
      >
        <div
          className="sc-iHGNiK bYqDie"
        >
          <div
            className="sc-dtBeHJ TZQnX"
          >
            <img
              className="sc-kOHUsU kSEzgl"
              src={Object {}}
              style={
                Object {
                  "border": "5px solid #5E17EB",
                }
              }
            />
            <h3>
              Emily Buckley
            </h3>
             Senior Engineer
          </div>
          <div
            className="sc-cPiJYC gOesTb sc-dtImxT jROJoV"
          >
            <a
              href="https://www.linkedin.com/in/emily-carter-buckley-a91a7118a/"
              target="blank"
            >
              <svg
                fill="#fff"
                height="35"
                viewBox="0 0 24 24"
                width="35"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-2 16h-2v-6h2v6zm-1-6.891c-.607 0-1.1-.496-1.1-1.109 0-.612.492-1.109 1.1-1.109s1.1.497 1.1 1.109c0 .613-.493 1.109-1.1 1.109zm8 6.891h-1.998v-2.861c0-1.881-2.002-1.722-2.002 0v2.861h-2v-6h2v1.093c.872-1.616 4-1.736 4 1.548v3.359z"
                />
              </svg>
            </a>
            <a
              href="https://github.com/emilybuc"
              target="blank"
            >
              <svg
                fill="#fff"
                height="35"
                viewBox="0 0 24 24"
                width="35"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm2.218 18.616c-.354.069-.468-.149-.468-.336v-1.921c0-.653-.229-1.079-.481-1.296 1.56-.173 3.198-.765 3.198-3.454 0-.765-.273-1.389-.721-1.879.072-.177.312-.889-.069-1.853 0 0-.587-.188-1.923.717-.561-.154-1.159-.231-1.754-.234-.595.003-1.193.08-1.753.235-1.337-.905-1.925-.717-1.925-.717-.379.964-.14 1.676-.067 1.852-.448.49-.722 1.114-.722 1.879 0 2.682 1.634 3.282 3.189 3.459-.2.175-.381.483-.444.936-.4.179-1.413.488-2.037-.582 0 0-.37-.672-1.073-.722 0 0-.683-.009-.048.426 0 0 .46.215.777 1.024 0 0 .405 1.25 2.353.826v1.303c0 .185-.113.402-.462.337-2.782-.925-4.788-3.549-4.788-6.641 0-3.867 3.135-7 7-7s7 3.133 7 7c0 3.091-2.003 5.715-4.782 6.641z"
                />
              </svg>
            </a>
          </div>
          <p
            className="sc-kOPcje kehCWa"
          >
            Hey there! I'm Emily, ðŸ˜Š an experienced Software Engineer, boasting a solid 5+ years track record at LEGO and RS Components. My passion lies in tackling challenges head-on and crafting innovative solutions that make a difference. Beyond work, I love to unwind by immersing myself in the captivating world of board games, keeping fit at the gym, and playing with my cat Hugo ðŸ˜º
          </p>
        </div>
        <div
          className="sc-iHGNiK bYqDie"
        >
          <div
            className="sc-dtBeHJ TZQnX"
          >
            <img
              className="sc-kOHUsU kSEzgl"
              src={Object {}}
              style={
                Object {
                  "border": "5px solid #F1A1E7",
                }
              }
            />
            <h3>
              Gregory Carter
            </h3>
             Junior Engineer
          </div>
          <div
            className="sc-cPiJYC gOesTb sc-dtImxT jROJoV"
          >
            <a
              href="https://www.linkedin.com/in/gregamo/"
              target="blank"
            >
              <svg
                fill="#fff"
                height="35"
                viewBox="0 0 24 24"
                width="35"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-2 16h-2v-6h2v6zm-1-6.891c-.607 0-1.1-.496-1.1-1.109 0-.612.492-1.109 1.1-1.109s1.1.497 1.1 1.109c0 .613-.493 1.109-1.1 1.109zm8 6.891h-1.998v-2.861c0-1.881-2.002-1.722-2.002 0v2.861h-2v-6h2v1.093c.872-1.616 4-1.736 4 1.548v3.359z"
                />
              </svg>
            </a>
            <a
              href="https://github.com/Gregamoo"
              target="blank"
            >
              <svg
                fill="#fff"
                height="35"
                viewBox="0 0 24 24"
                width="35"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm2.218 18.616c-.354.069-.468-.149-.468-.336v-1.921c0-.653-.229-1.079-.481-1.296 1.56-.173 3.198-.765 3.198-3.454 0-.765-.273-1.389-.721-1.879.072-.177.312-.889-.069-1.853 0 0-.587-.188-1.923.717-.561-.154-1.159-.231-1.754-.234-.595.003-1.193.08-1.753.235-1.337-.905-1.925-.717-1.925-.717-.379.964-.14 1.676-.067 1.852-.448.49-.722 1.114-.722 1.879 0 2.682 1.634 3.282 3.189 3.459-.2.175-.381.483-.444.936-.4.179-1.413.488-2.037-.582 0 0-.37-.672-1.073-.722 0 0-.683-.009-.048.426 0 0 .46.215.777 1.024 0 0 .405 1.25 2.353.826v1.303c0 .185-.113.402-.462.337-2.782-.925-4.788-3.549-4.788-6.641 0-3.867 3.135-7 7-7s7 3.133 7 7c0 3.091-2.003 5.715-4.782 6.641z"
                />
              </svg>
            </a>
          </div>
          <p
            className="sc-kOPcje kehCWa"
          >
            Hey, I'm Greg! ðŸ˜„ I trained as a video game designer but pursued a career in software engineering, where I thrive on solving complex technical challenges. My game design background fuels my creativity and problem-solving skills in the tech world. Outside of work, I love spending time with my daughter, attending live concerts, playing video games, and reading manga. I also enjoy going to the gym when I have time. This blend of technical expertise and diverse interests keeps me inspired and engaged in all aspects of my life.ðŸŒŸ
          </p>
        </div>
      </div>
    </div>
  </div>
  <div
    className="sc-dLNtp kZvCms"
  >
    <div
      className="sc-cwHqhk fuAgyt sc-eDPFhE easHTf"
    >
      <img
        alt="Fuji Digital Logo"
        className="sc-dAlxHm iJGhYI"
        src="fuji-digital-logo.svg"
      />
      <img
        alt="Fuji Digital Text Logo"
        className="sc-jlZhRR eJgNbx"
        data-testid="textLogo"
        src="fuji-digital-text-logo-purple.svg"
      />
    </div>
    <div
      className="sc-eldOKa eGsMBu"
    >
      <div>
        Fuji Digital Ltd.
      </div>
      <a
        href="mailto: contact@fujidigital.co.uk"
      >
        contact@fujidigital.co.uk
      </a>
    </div>
    <div
      className="sc-cPiJYC gOesTb sc-fPXMhL bwLCFr"
    >
      <a
        href="https://www.linkedin.com/company/fuji-digital-tech/"
        target="blank"
      >
        <svg
          height="35"
          viewBox="0 0 24 24"
          width="35"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-2 16h-2v-6h2v6zm-1-6.891c-.607 0-1.1-.496-1.1-1.109 0-.612.492-1.109 1.1-1.109s1.1.497 1.1 1.109c0 .613-.493 1.109-1.1 1.109zm8 6.891h-1.998v-2.861c0-1.881-2.002-1.722-2.002 0v2.861h-2v-6h2v1.093c.872-1.616 4-1.736 4 1.548v3.359z"
          />
        </svg>
      </a>
      <a
        href="https://github.com/fuji-digital"
        target="blank"
      >
        <svg
          height="35"
          viewBox="0 0 24 24"
          width="35"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 0c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm2.218 18.616c-.354.069-.468-.149-.468-.336v-1.921c0-.653-.229-1.079-.481-1.296 1.56-.173 3.198-.765 3.198-3.454 0-.765-.273-1.389-.721-1.879.072-.177.312-.889-.069-1.853 0 0-.587-.188-1.923.717-.561-.154-1.159-.231-1.754-.234-.595.003-1.193.08-1.753.235-1.337-.905-1.925-.717-1.925-.717-.379.964-.14 1.676-.067 1.852-.448.49-.722 1.114-.722 1.879 0 2.682 1.634 3.282 3.189 3.459-.2.175-.381.483-.444.936-.4.179-1.413.488-2.037-.582 0 0-.37-.672-1.073-.722 0 0-.683-.009-.048.426 0 0 .46.215.777 1.024 0 0 .405 1.25 2.353.826v1.303c0 .185-.113.402-.462.337-2.782-.925-4.788-3.549-4.788-6.641 0-3.867 3.135-7 7-7s7 3.133 7 7c0 3.091-2.003 5.715-4.782 6.641z"
          />
        </svg>
      </a>
    </div>
    <div
      className="sc-gFqAYk jmzJaC"
    >
      Copyright Â© 2024 Fuji Digital Ltd. All rights reserved.
    </div>
    <img
      className="sc-ikkyvV ffTpDd"
      src="white-purple-decorative-blobs.svg"
    />
    <img
      className="sc-dAbbbq dBpaqa"
      src="pink-round-tall-fujit.svg"
    />
  </div>
</div>
`;
